### PR TITLE
buildcache: fix error handling on failed list

### PR DESCRIPTION
**Pull request by becker33**
Error reported on slack.

```
[imjelde@lblhcbpr20 spack]$ spack buildcache list
==> Error: Multiple errors during fetching:
        Error 1: RuntimeError: Unable to read index hash https://mirror.spack.io/build_cache/index.json.hash due to SpackWebError: Download failed: HTTP Error 404: Not Found
        Error 2: RuntimeError: Unable to read index https://mirror.spack.io/build_cache/index.json due to SpackWebError: Download failed: HTTP Error 404: Not Found
==> Error: local variable 'specs' referenced before assignment
```

This fixes the variable reference error, not the underlying buildcache access error.

### CODE REVIEW

File: lib/spack/spack/cmd/buildcache.py [hunk 1]
```diff
@@ -364,6 +364,7 @@ def list_fn(args):
     try:
         specs = bindist.update_cache_and_get_specs()
     except bindist.FetchCacheError as e:
+        specs = []
```
Comment (alalazo) [inline lib/spack/spack/cmd/buildcache.py:367]: ~Should we `tty.die` instead?~ I'm getting back to this. Considering that the most common failure is very likely on our source mirror, we should probably do the opposite of what I was proposing and turn `tty.error` into `tty.debug`. 

This would just be a stopgap solution since a better way to deal with the issue is a refactor of the methods updating caches in `spack.binary_distribution`. For instance, currently:
1. We get an error only if all the fetches from all the mirrors failed
2. We might get a false positive if the mirrors are just source mirrors i.e. they don't have buildcaches

The two point above should probably change, to log what Spack is doing and which mirror failed to provide an `index.json` and to avoid the false positives for source only mirrors. But that might be for later PRs.

---

**TL;DR** I think we should change the `tty.error` below to `tty.debug`

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a minor request and a few considerations in https://github.com/spack/spack/pull/30045#discussion_r850263715

**Comment by alalazo** [comment]
Outdated since now the program uses `tty.die(e)`.
